app/uk/gov/hmrc/leakdetection/services/GithubService.scala

```diff
@@ -29,23 +29,15 @@ class GithubService @Inject()(httpClient: HttpClient, configLoader: ConfigLoader
 
   import configLoader.cfg
 
-  def getDefaultBranchName(repository: String)(implicit ec: ExecutionContext, hc: HeaderCarrier): Future[String] = {
+  def getDefaultBranchName(repository: String)(implicit ec: ExecutionContext, hc: HeaderCarrier): Future[Branch] = {
     val githubAccessToken = cfg.githubSecrets.personalAccessToken
     val url = s"${cfg.github.apiUrl}/$repository"
-    httpClient.GET[Option[RepoInfo]](
+    httpClient.GET[Option[Branch]](
       url = url,
       headers = Seq(("Authorization", s"token $githubAccessToken"))) recover {
-      case _ => None
+      case _ => Some(Branch("main"))
```

Comment:
can we make the default a constant on Branch object (`Branch.main` or `Branch.default`)?

Reply:
I think the recover should be removed - better to fail if there's an error than return a misleading result.
If it is kept, it should probably return `None` (and log the error), so that the fallback logic is not duplicated.